src/engine/renderer/tr_shader.cpp

```diff
@@ -2641,6 +2659,14 @@ static bool ParseStage( shaderStage_t *stage, const char **text )
 				depthMaskBits = 0;
 			}
 		}
+		else if ( !Q_stricmp( token, "linearColorMap" ) )
```

Comment:
I dislike the "linear X" naming. It suggests that the input is in a linear color space and so it should be converted to whatever colorspace is used for further operations. But the point is rather that the resulting color is deliberately made different depending on the blend regime by not converting. So I suggest a name that describes it more as disabling color conversion, like `noTextureColorspaceConversion`.

Also for the case of the color uniform, maybe we could make it more obvious what the modifier is used for by making it part of the rgbgen syntax. Something like `rgbgen noconvert const (0.5 0.5 0.5)`

Reply:
> It suggests that the input is in a linear color space

That suggestion is correct. That keyword says the image data should be considered linear.

> so it should be converted to whatever colorspace is used for further operations

It's just that we have no reason to convert to another colorspace here, but if we had a reason to, it would do it, yes.

Reply:
As I understand it, currently the rgbgen values are always understood as sRGB values. So when rendering in the sRGB-ish naive regime, the values are used as-is; in the linear blending regime, they must be converted to linear space. Conversely, I expect that if some rgbgen values were to be understood as linear, then they would be used as-is in linear mode, but need conversion with naive blending. Instead, the feature proposed here is to treat the values as being whatever colorspace the renderer is currently running with, as a dirty hack to make some colors brighter in linear mode. With the proposed code, colors are NOT treated as linear when using the naive blending regime, so using a keyword named `linear` is wrong/misleading.

Reply:
Those proposed keywords have no meaning when rendering a non colorspace-aware map to begin with.

Reply:
There is no naive blending use case for those keywords.

Reply:
Yes, but it works by being unused in naive mode.